src/qt/treectrl.cpp

```diff
@@ -56,275 +495,699 @@ void wxTreeCtrl::SetIndent(unsigned int indent)
 {
-    return wxTreeItemId();
+    wxCHECK_MSG(item.IsOk(), wxTreeItemId(), "invalid tree item");
+
+    const QTreeWidgetItem *qTreeItem = wxQtConvertTreeItem(item);
+    const int childCount = qTreeItem->childCount();
+    return childCount == 0 
+        ? wxTreeItemId()
+        : wxQtConvertTreeItem(qTreeItem->child(childCount - 1));
 }
 
 wxTreeItemId wxTreeCtrl::GetNextSibling(const wxTreeItemId& item) const
 {
-    return wxTreeItemId();
+    wxCHECK_MSG(item.IsOk(), wxTreeItemId(), "invalid tree item");
+
+    QTreeWidgetItem *qTreeItem = wxQtConvertTreeItem(item);
+    QTreeWidgetItem *parent = qTreeItem->parent();
+
+    if ( parent != NULL )
+    {
+        int index = parent->indexOfChild(qTreeItem);
+        wxASSERT(index != -1);
+
+        ++index;
+        return index < parent->childCount()
+            ? wxQtConvertTreeItem(parent->child(index))
+            : wxTreeItemId();
+    }
+
+    int index = m_qtTreeWidget->indexOfTopLevelItem(qTreeItem);
+    wxASSERT(index != -1);
+
+    ++index;
+    return index < m_qtTreeWidget->topLevelItemCount()
+        ? wxQtConvertTreeItem(m_qtTreeWidget->topLevelItem(index)) 
+        : wxTreeItemId();
 }
 
 wxTreeItemId wxTreeCtrl::GetPrevSibling(const wxTreeItemId& item) const
 {
-    return wxTreeItemId();
+    wxCHECK_MSG(item.IsOk(), wxTreeItemId(), "invalid tree item");
+
+    QTreeWidgetItem *qTreeItem = wxQtConvertTreeItem(item);
+    QTreeWidgetItem *parent = qTreeItem->parent();
+
+    if ( parent != NULL )
+    {
+        int index = parent->indexOfChild(qTreeItem);
+        wxASSERT(index != -1);
+
+        --index;
+        return index >= 0
+            ? wxQtConvertTreeItem(parent->child(index))
+            : wxTreeItemId();
+    }
+
+    int index = m_qtTreeWidget->indexOfTopLevelItem(qTreeItem);
+    wxASSERT(index != -1);
+
+    --index;
+    return index >= 0
+        ? wxQtConvertTreeItem(m_qtTreeWidget->topLevelItem(index))
+        : wxTreeItemId();
 }
 
 wxTreeItemId wxTreeCtrl::GetFirstVisibleItem() const
 {
+   wxTreeItemId itemid = GetRootItem();
+    if ( !itemid.IsOk() )
+        return itemid;
+
+    do
+    {
+        if (IsVisible(itemid))
+            return itemid;
+        itemid = GetNext(itemid);
+    } while ( itemid.IsOk() );
+
     return wxTreeItemId();
 }
 
 wxTreeItemId wxTreeCtrl::GetNextVisible(const wxTreeItemId& item) const
 {
+    wxCHECK_MSG(item.IsOk(), wxTreeItemId(), wxT("invalid tree item"));
+    wxASSERT_MSG(IsVisible(item), wxT("this item itself should be visible"));
+
+    wxTreeItemId id = item;
+    if ( id.IsOk() )
+    {
+        while ( id = GetNext(id), id.IsOk() )
+        {
+            if ( IsVisible(id) )
+                return id;
+        }
+    }
     return wxTreeItemId();
 }
 
 wxTreeItemId wxTreeCtrl::GetPrevVisible(const wxTreeItemId& item) const
 {
-    return wxTreeItemId();
+    wxCHECK_MSG(item.IsOk(), wxTreeItemId(), wxT("invalid tree item"));
```

Comment:
Please avoid `wxT()` in the new code. If it's very performance-sensitive you can use `wxS()` but usually it's just fine to not use any macros:
```suggestion
    wxCHECK_MSG(item.IsOk(), wxTreeItemId(), "invalid tree item");
```

Reply:
Fixed